std/assembly/util/number.ts

```diff
@@ -716,28 +726,47 @@ function prettify(buffer: usize, length: i32, k: i32): i32 {
   }
 }
 
-function dtoa_core(buffer: usize, value: f64): i32 {
+function dtoa_core(buffer: usize, value: f64, isSinglePrecision: bool): i32 {
   let sign = i32(value < 0);
   if (sign) {
     value = -value;
     store<u16>(buffer, CharCode.MINUS);
   }
-  // assert(value > 0 && value <= 1.7976931348623157e308);
-  let len = grisu2(value, buffer, sign);
+  // assert(value > 0 && value <= (isSinglePrecision ? f32.MAX_VALUE : f64.MAX_VALUE));
+  let len = grisu2(value, buffer, sign, isSinglePrecision);
   len = prettify(buffer + (sign << 1), len - sign, _K);
   return len + sign;
 }
 
 // @ts-ignore: decorator
 @lazy @inline const dtoa_buf = memory.data(MAX_DOUBLE_LENGTH << 1);
 
-export function dtoa(value: f64): String {
+export function dtoa<T extends number>(value: T): String {
+  // assert(isFloat<T>());
+  if (sizeof<T>() == 4) {
+    return dtoa_impl(<f32>value, true);
+  } else {
+    return dtoa_impl(<f64>value, false);
+  }
+}
+
+export function dtoa32(value: f32): String {
+  return dtoa_impl(value, true);
+}
+
+export function dtoa64(value: f64): String {
+  return dtoa_impl(value, false);
+}
```

Comment:
Not sure if we need these two extra methods while `dtoa` is polymorphic now (`dtoa<T extends number>(value: T)`)

Reply:
I thought it would be more optimal, but it's easily removed if you prefer.  Here's a commit on a separate branch with just that change so you can see the WAT differences.
https://github.com/mattjohnsonpint/assemblyscript/commit/55f1789935dc719ef6f5ab24a321f7cfa98a5c7a

LMK which you prefer.  Thanks.

Reply:
interestingly the only release wat that changed was the one for this issue.  The others optimized to the same wat.

Reply:
I think better to remove these and keep only single dtoa